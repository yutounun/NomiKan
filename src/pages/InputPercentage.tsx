import React, { useState } from "react";
import { Stack } from "@mui/material";

import BaseLayout from "components/Commons/Layout/BaseLayout";
import MyHeader from "components/Commons/Organisms/MyHeader";
import MyStepper from "components/Commons/Organisms/MyStepper";
import Content from "components/InputPercentage/Organisms/Content";

// import MyAlert from "components/Commons/Molecules/MyAlert";

function AddMembers() {
  const [open, setOpen] = useState(false);
  const [alertLabel, setAlertLabel] = useState("");

  console.log(alertLabel, open);

  // const handleCloseAlert = () => {
  //   setOpen(false);
  // };

  return (
    <BaseLayout>
      <Stack direction="column" justifyContent="center" alignItems="center">
        {/* 割合入力Header */}
        <MyHeader value="割合入力" />

        {/* Stepper showing the current page */}
        <MyStepper activeStep={2} />

        {/* Content including input and button */}
        <Content setOpen={setOpen} setAlertLabel={setAlertLabel} />

        {/* Alert for the completion of registration  */}
        {/* <MyAlert parentOpen={open} value={alertLabel} onClose={handleCloseAlert} /> */}

      </Stack>
    </BaseLayout>
  );
}

export default AddMembers;
