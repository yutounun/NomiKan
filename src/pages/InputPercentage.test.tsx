import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import InputPercentage from "pages/InputPercentage";

/**
 * Requirements
 * Renders a component in the DOM x
 * Renders MyStepper component with the current step x
 * Renders add button component x
 * Disable the registration button, if none of members's percentage is changed x
 * Renders name, percentage, and edit button on the Member component x
 * Opens Edit Member modal on clicking edit button x
 * Renders typed percentage on the DOM after editing on modal component works x
 * Disable the registration button, when the total percentage is more than 100%
 * Disable the registration button, if the amount of the total percentage is not 100%
 * Shows "割合入力" on the title x
 */

describe("InputPercentage page", () => {
  beforeEach(() => {
    sessionStorage.clear();
  });

  it("renders correctly", () => {
    render(
      <Router>
        <InputPercentage />
      </Router>
    );
  });

  it("takes a snapshot", () => {
    const { asFragment } = render(
      <Router>
        <InputPercentage />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Shows 割合入力 on the title", () => {
    render(
      <Router>
        <InputPercentage />
      </Router>
    );

    const header = screen.getByRole("heading", {
      name: /ratio/i,
    });

    expect(header).toBeInTheDocument();
  });

  it("Renders MyStepper component with the current step", () => {
    render(
      <Router>
        <InputPercentage />
      </Router>
    );
    const totalAmountStep = screen.getByTestId("total amount");
    // Only inactive steps have "Mui-disabled" class
    expect(totalAmountStep).not.toHaveClass("Mui-disabled");

    const addMemberStep = screen.getByTestId("members");
    expect(addMemberStep).not.toHaveClass("Mui-disabled");

    const inputPercentage = screen.getByTestId("ratio");
    expect(inputPercentage).not.toHaveClass("Mui-disabled");
  });

  it("Renders add button component", () => {
    render(
      <Router>
        <InputPercentage />
      </Router>
    );
    const textField = screen.getByRole("button", {
      name: /Submit/i
    });
    expect(textField).toBeInTheDocument();
  });

  it("Disable the registration button initially", () => {
    render(
      <Router>
        <InputPercentage />
      </Router>
    );

    // Make sure the registration button is disabled
    expect(screen.getByText("Submit")).toBeDisabled();
  });

  it("has a registration button component", () => {
    render(
      <Router>
        <InputPercentage />
      </Router>
    );
    const regBtn = screen.getByRole("button", {
      name: /Submit/i
    });
    expect(regBtn).toBeInTheDocument();
  });

  // it("Renders name, percentage, and edit button on the Member component", () => {
  //   window.sessionStorage.setItem(
  //     "nomikan-storage",
  //     JSON.stringify({
  //       state: { cost: "100", members: [{ name: "Yuto", ratio: 0 }] },
  //       version: 0,
  //     })
  //   );

  // it("Renders name, percentage, and edit button on the Member component", () => {
  //   window.sessionStorage.setItem(
  //     "nomikan-storage",
  //     JSON.stringify({
  //       state: { cost: "100", members: [{ name: "Yuto", ratio: 0 }] },
  //       version: 0,
  //     })
  //   );

  //   render(
  //     <Router>
  //       <InputPercentage />
  //     </Router>
  //   );

  // screen.debug(undefined, Infinity);
  // render(
  //   <Router>
  //     <InputPercentage />
  //   </Router>
  // );
  // const memberName = screen.getByText("Yuto");
  // expect(memberName).toBeInTheDocument();
  // });

  // it("Opens Edit Member modal on clicking edit button", () => {
  //   sessionStorage.setItem(
  //     "nomikan-storage", '{"state":{"cost":"","members":["Yuto"]},"version":0}'
  //   );
  //   render(
  //   <Router>
  //     <InputPercentage />
  //   </Router>
  // );

  //   // Click add button
  //   fireEvent(
  //     screen.getByTestId("edit-btn"),
  //     new MouseEvent("click", {
  //       bubbles: true,
  //       cancelable: true,
  //     })
  //   );

  //   const editModalTitle = screen.getByText("編集");
  //   expect(editModalTitle).toBeInTheDocument();
  // });

  // it("Renders typed percentage on the DOM after editing on modal component works", () => {
  //   sessionStorage.setItem(
  // "nomikan-storage", '{"state":{"cost":"","members":["Yuto"]},"version":0}'
  // );
  //   render(
  //   <Router>
  //     <InputPercentage />
  //   </Router>
  // );

  //   // Click add button
  //   fireEvent(
  //     screen.getByTestId("edit-btn"),
  //     new MouseEvent("click", {
  //       bubbles: true,
  //       cancelable: true,
  //     })
  //   );

  //   screen.getByText("編集");

  //   // Type on TextField component using fireEvent
  //   const textField = screen.getByRole("textbox", {
  //     name: /%/i
  //   });
  //   fireEvent.change(textField, { target: { value: "40" } });

  //   // close the modal
  //   fireEvent(
  //     screen.getByTestId("complete-btn"),
  //     new MouseEvent("click", {
  //       bubbles: true,
  //       cancelable: true,
  //     })
  //   );

  //   screen.getByText("40%");
  // });
});
