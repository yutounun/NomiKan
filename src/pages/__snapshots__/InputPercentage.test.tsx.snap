// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPercentage page takes a snapshot 1`] = `
<DocumentFragment>
  <main
    data-testid="main"
  >
    <div
      class="MuiStack-root css-w004l2-MuiStack-root"
    >
      <div
        class="MuiStack-root css-muz5cb-MuiStack-root"
        data-testid="header"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-vh66nq-MuiTypography-root"
        >
          割合入力
        </h1>
      </div>
      <div
        class="MuiStack-root css-1rh9169-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root css-pb02ub"
      >
        <div
          class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-10mg1vw-MuiStepper-root"
          data-testid="stepper"
        >
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel Mui-completed css-n7tliy-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
              data-testid="合計金額"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-completed MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-completed MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    合計金額
                  </p>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel Mui-completed css-n7tliy-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-completed css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
              data-testid="メンバー追加"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-completed MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-completed MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    メンバー追加
                  </p>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-active css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
              data-testid="割合入力"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-active MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    割合入力
                  </p>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
              data-testid="結果"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    4
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    結果
                  </p>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root css-14u1duc-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1fcghb4-MuiStack-root"
        />
        <a
          data-testid="reg-btn"
          href="/result"
          style="text-decoration: none; width: 100%; text-align: center;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ei57pa-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DoneOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m19.77 5.03 1.4 1.4L8.43 19.17l-5.6-5.6 1.4-1.4 4.2 4.2L19.77 5.03m0-2.83L8.43 13.54l-4.2-4.2L0 13.57 8.43 22 24 6.43 19.77 2.2z"
                />
              </svg>
            </span>
            登録
          </button>
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
